SDK/AppCenter/Microsoft.AppCenter.Windows.Shared/Storage/Storage.cs

```diff
@@ -85,14 +86,57 @@ public Task PutLog(string channelName, Log log)
             return AddTaskToQueue(() =>
             {
                 var logJsonString = LogSerializer.Serialize(log);
-                _storageAdapter.Insert(TableName,
-                    new[] { ColumnChannelName, ColumnLogName },
-                    new List<object[]> {
-                        new object[] {channelName, logJsonString}
-                    });
+                var maxSize = _storageAdapter.GetMaxStorageSize();
+                var logSize = Encoding.UTF8.GetBytes(logJsonString).Length;
+                if (maxSize == -1)
+                {
+                    throw new StorageException("Failed to store a log to the database.");
+                }
+                if (maxSize <= logSize)
```

Comment:
I think it's supposed to be `maxSize - <usedSize> <= logSize`

Reply:
This PR was done for this scenario (case when we have no free space for a new log). If database size is enough to insert the log but we have not enough free space then we should free space for this log. If the log size is more then database size then we don't need to insert this log and we don't need to free database space.